### ISSUE TITLE
Finish adapting build-server for phet-io

### ISSUE THREAD

**Issue by mattpen** [phetsims/perennial#26]
- [x] .htaccess should be copied to /SIM/VERSION/protected, it is currently copied to /SIM/VERSION/
- [x] rc deploys are not completing the task and are causing subsequent builds to queue indefinitely
- [x] Initial production deploys are consistently failing due to repos failing to pull master.  Subsequent deploys complete successfully.

Changes are tracked in the `phet-io-changes` branch.

**Comment by mattpen** [comment]
> Initial production deploys are consistently failing due to repos failing to pull master. Subsequent deploys complete successfully.

This appears to happen if the build-server is started and there are repos who are not checked out to master.  For example if the build-server hangs or crashes before calling afterDeploy.  I'm going to add a `grunt checkout-master` call to the initialization of the app.

**Comment by mattpen** [comment]
@jbphet - These changes should be stable and have been tested on phet-server.  Would you like to review the changes before I merge them in with the master branch?

**Comment by jbphet** [comment]
@mattpen said:

> I'm going to add a grunt checkout-master call to the initialization of the app.

I'm looking at the changes, and it doesn't look like that was what was done.  It looks like instead a checkout of master was added to the method ```pullMaster```.  This makes sense, and seems like a good way to fix the problem that was observed, but thought I should note it.  It looks like other code cleanup was also done.

Other changes have been done to ```build-server.js``` since the commits shown above.  The commits above look good, but before merging all of the changes into master someone (and I'd be glad to do it) should do a final review by diffing the ```phet-io-changes``` branch with the master branch.

**Comment by mattpen** [comment]
@jbphet 

> It looks like instead a checkout of master was added to the method pullMaster

Sam and I worked on this together and decided this would be better as it was somewhat ambiguous which `dependencies.json` would be used if we did `grunt checkout-master`.  We tested it out and it appears to be working nicely.

I'll review the results of the next RC deploy for the changes in phetsims/perennial#20 and reassign for final review/merging if everything looks good.
